### ISSUE TITLE
when  a database is empty  , it will not be restored

### ISSUE THREAD

**Issue by wangbinhe3db** [pingcap/br#828]
## Question

		if len(dbInfo.Tables) == 0 {
			log.Warn("It's not necessary for backing up empty database",
				zap.Stringer("db", dbInfo.Name))
			continue
		}

**Comment by kennytm** [comment]
cc #337.

**Comment by wangbinhe3db** [comment]
@kennytm   when a empty database is used  backup and restore ,       i have nothing to do ?

**Comment by wangbinhe3db** [comment]
@YuJuncen hi  could I pull a PR  code to  accomplish this

**Comment by YuJuncen** [comment]
> @YuJuncen hi could I pull a PR code to accomplish this

Of course. Due to the `Schema` structure (see below), maybe leave `table` field empty would be OK. But this may introduce new compatibility problems. (e.g. Backups with newer version BR and this flag cannot be restored in an older version of BR), is that OK? @overvenus @kennytm

**Comment by wangbinhe3db** [comment]
yep   i  think    if backup  empty database.  then should restore  empty database, if  not,  should not restore  empty database.

**Comment by YuJuncen** [comment]
@wangbinhe3db Yes, and the current problem is, if we generate a backupmeta with empty databases, I'm afraid that older versions of BR might have no way to recognize them.

**Comment by kennytm** [comment]
i think the following behavior is acceptable:

* **old** backup → **any** restore = empty db not backup, and thus skipped
* **new** backup → **old** restore = creating empty db is "optional": the old BR can either ignore the new field (thanks to protobuf) or actually create the empty db via DDL.
* **new** backup → **new** restore = create the empty db.

**Comment by YuJuncen** [comment]
@kennytm You mean, following https://github.com/pingcap/br/issues/337#issuecomment-640359151 or https://github.com/pingcap/br/issues/337#issuecomment-640363405?

**Comment by kennytm** [comment]
the second one.

**Comment by YuJuncen** [comment]
@wangbinhe3db And, eh, I'm not sure whether I fully understand what you are wanting us to do. @kennytm's solution has been ready to be working with. 

If you meaning you want to provide a PR for us(thanks for that!), I can give you a recipe then. 

Or you're just want us to give a hot fix? If so, I'm ready to do that.

**Comment by wangbinhe3db** [comment]
I   want to provide a PR   for community @YuJuncen

**Comment by YuJuncen** [comment]
@wangbinhe3db OK, so as #337 said, current version of backupmeta doesn't support backing up empty schemas at all. So, firstly, you should update the backupmeta at the `[kvproto](https://GitHub.com/pingcap/kvproto)` repository:
